src/ui/render.ts

```diff
@@ -18,15 +19,49 @@ export default class UI {
     this.rowCount = rowCount;
     this.lightSheet = lightSheet;
 
+    this.tableEl.classList.add("light_sheet_table_container");
+
+    const lightSheetContainerDom = document.createElement("div");
+    lightSheetContainerDom.classList.add("light_sheet_table_content");
+    this.tableEl.appendChild(lightSheetContainerDom);
+
     const tableContainerDom = document.createElement("table");
-    this.tableEl.appendChild(tableContainerDom);
+    tableContainerDom.classList.add("light_sheet_table");
+    tableContainerDom.setAttribute("cellpadding", "0");
+    tableContainerDom.setAttribute("cellspacing", "0");
+    tableContainerDom.setAttribute("unselectable", "yes");
+    lightSheetContainerDom.appendChild(tableContainerDom);
+
+    //thead
+    this.tableHeadDom = document.createElement("thead");
+    tableContainerDom.appendChild(this.tableHeadDom);
+
+    //tbody
     this.tableBodyDom = document.createElement("tbody");
     tableContainerDom.appendChild(this.tableBodyDom);
   }
 
-  addRow(): Element {
+  addHeader(headerData: string[]) {
+    const headerRowDom = document.createElement("tr");
+    this.tableHeadDom.appendChild(headerRowDom);
+
+    for (let i = 0; i < headerData.length; i++) {
+      const headerCellDom = document.createElement("td");
+      headerCellDom.textContent = headerData[i];
+      headerRowDom.appendChild(headerCellDom);
+    }
+  }
+
+  addRow(i: number): Element {
```

Comment:
change the variable name from i to somethis more readable, maybe rowIndex/rowNumber

Reply:
I changed "i" to "rowLabelNumber"